features/support/env.rb

```diff
@@ -81,3 +81,27 @@ def output_is_script_readable(raw_stdout)
 Then(/^stdout from "([^"]*)" should match \/(.*)\/$/) do |cmd, regexp|
   aruba.command_monitor.find(Aruba.platform.detect_ruby(cmd)).send(:stdout) =~ /#{regexp}/
 end
+
+Then(/^service "([^"]*)" should be ([^"]*)$/) do |service, operation|
+  run("vagrant ssh -c \"sudo systemctl status docker\"")
+
+  if operation == 'started'
+    exit_code = 0
+    regexp = /Active: active \(running\)/
+  elsif operation == 'stopped'
+    exit_code = 3
+    regexp = /Active: inactive\(dead\)/
+  end
+
+  expect(last_command_started).to have_exit_status(exit_code)
+  aruba.command_monitor.find(Aruba.platform.detect_ruby(last_command_started)).send(:stdout) =~ regexp
+end
+
+Then(/^command "([^"]*)" should restart "([^"]*)"$/) do |cmd, service|
+  run("vagrant ssh -c \"sudo systemctl status docker\"")
+  pid = last_command_started.pid
```

Comment:
I don't think this is right at all. The pids you are comparing are the pids of the commands you are executing. These are not the pids of the docker daemon. You need to get the docker daemon pid from the output of systemctl status.

You most likely need to store the pid in a instance variable:

    When the "docker" service is running

The step impl for this would verify that the service is indeed running and store its pid in `@service_pid`.

    Then the service "docker" should have restarted

In the step impl here you would again verify that the servive is running and extract its pid (current_pid). Then you assert @service_pid != current_pid.